File: skills/architecture/committee-decision-making/examples/python/basic.py
#!/usr/bin/env python3
"""
Basic Example: Committee Decision Making (Python)

This example shows the simplest way to use the committee decision-making skill.

Requirements:
    pip install requests
"""

import requests
import time
import json
import sys

SERVER_URL = "http://localhost:8000/mcp"

def start_session(topic, participants, rounds=1):
    """Start a committee discussion session."""
    response = requests.post(
        f"{SERVER_URL}/tools/start_committee_session",
        json={
            "topic": topic,
            "participants": participants,
            "debate_rounds": rounds
        }
    )
    response.raise_for_status()
    return response.json()

def get_status(session_id):
    """Get session status."""
    response = requests.post(
        f"{SERVER_URL}/tools/get_session_status",
        json={"session_id": session_id}
    )
    response.raise_for_status()
    return response.json()

def get_synthesis(session_id):
    """Get discussion synthesis."""
    response = requests.post(
        f"{SERVER_URL}/tools/synthesize_discussion",
        json={"session_id": session_id}
    )
    response.raise_for_status()
    return response.json()

def wait_for_completion(session_id, timeout=300):
    """Poll until session completes."""
    start_time = time.time()
    
    while time.time() - start_time < timeout:
        status = get_status(session_id)
        
        print(f"   Status: {status['status']}, Phase: {status.get('current_phase', 'N/A')}")
        
        if status['status'] == 'completed':
            return True
        elif status['status'] == 'error':
            raise Exception("Session failed")
        
        time.sleep(5)
    
    raise TimeoutError(f"Session did not complete within {timeout} seconds")

def main():
    print("🚀 Starting basic committee discussion...\n")
    
    # Start session
    result = start_session(
        topic="Should we adopt microservices architecture?",
        participants=[
            {"role": "architect", "provider": "openai", "model": "gpt-4"},
            {"role": "devops", "provider": "anthropic", "model": "claude-3-sonnet"}
        ],
        rounds=1
    )
    
    session_id = result['session_id']
    print(f"📋 Session started: {session_id}\n")
    
    # Wait for completion
    print("⏳ Waiting for discussion to complete...\n")
    wait_for_completion(session_id)
    
    # Get synthesis
    print("\n📊 Getting synthesis...\n")
    synthesis = get_synthesis(session_id)
    
    # Display results
    print("=== Committee Decision ===\n")
    print("Summary:")
    print(synthesis['synthesis']['summary'])
    print("\nRecommendations:")
    for i, rec in enumerate(synthesis['synthesis']['recommendations'], 1):
        print(f"{i}. {rec}")
    
    print("\n✅ Done!")

if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(f"❌ Error: {e}", file=sys.stderr)
        sys.exit(1)

File: skills/committee-decision-making/examples/javascript/basic.js
#!/usr/bin/env node

/**
 * Basic Example: Committee Decision Making
 * 
 * This example shows the simplest way to use the committee decision-making skill.
 */

import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { SSEClientTransport } from "@modelcontextprotocol/sdk/client/sse.js";

const SERVER_URL = process.env.MCP_SERVER_URL || "http://localhost:8000/mcp";

async function main() {
  console.log("🚀 Starting basic committee discussion...\n");

  // 1. Connect to MCP server
  const client = new Client(
    { name: "basic-example", version: "1.0.0" },
    { capabilities: {} }
  );

  await client.connect(new SSEClientTransport(new URL(SERVER_URL)));
  console.log("✅ Connected to MCP server\n");

  // 2. Start committee session
  const result = await client.callTool({
    name: "start_committee_session",
    arguments: {
      topic: "Should we adopt microservices architecture?",
      participants: [
        { role: "architect", provider: "openai", model: "gpt-4" },
        { role: "devops", provider: "anthropic", model: "claude-3-sonnet" }
      ],
      debate_rounds: 1
    }
  });

  const sessionId = result.session_id;
  console.log(`📋 Session started: ${sessionId}\n`);

  // 3. Poll for completion
  console.log("⏳ Waiting for discussion to complete...\n");
  
  let status;
  do {
    await sleep(5000);
    status = await client.callTool({
      name: "get_session_status",
      arguments: { session_id: sessionId }
    });
    console.log(`   Status: ${status.status}, Phase: ${status.current_phase}`);
  } while (status.status !== "completed" && status.status !== "error");

  if (status.status === "error") {
    console.error("\n❌ Session failed");
    process.exit(1);
  }

  // 4. Get synthesis
  console.log("\n📊 Getting synthesis...\n");
  
  const synthesis = await client.callTool({
    name: "synthesize_discussion",
    arguments: { session_id: sessionId }
  });

  // 5. Display results
  console.log("=== Committee Decision ===\n");
  console.log("Summary:");
  console.log(synthesis.synthesis.summary);
  console.log("\nRecommendations:");
  synthesis.synthesis.recommendations.forEach((rec, i) => {
    console.log(`${i + 1}. ${rec}`);
  });

  console.log("\n✅ Done!");
}

function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

main().catch(error => {
  console.error("❌ Error:", error.message);
  process.exit(1);
});
